substrate/client/network/src/service.rs

```diff
@@ -1020,7 +1023,7 @@ where
 	}
 
 	fn report_peer(&self, peer_id: sc_network_types::PeerId, cost_benefit: ReputationChange) {
-		self.peer_store_handle.clone().report_peer(peer_id, cost_benefit);
+		self.peer_store_handle.report_peer(peer_id, cost_benefit);
```

Comment:
I wonder what was the purpose of that `clone()` and why it's not needed anymore?

Reply:
The peer store handle is behind an Arc `peer_store_handle: Arc<dyn PeerStoreProvider>`, therefore the cloning should be cheap here.  I think this was a miss, there's no need to clone the handler to access its methods (we do have other calls to report_peer which don't clone)